core/templates/pages/topic-editor-page/modal-templates/create-new-subtopic-modal.component.spec.ts

```diff
@@ -0,0 +1,254 @@
+import { Subtopic } from 'domain/topic/subtopic.model';
+import { EventEmitter, NO_ERRORS_SCHEMA } from '@angular/core';
+import { SubtopicPage } from 'domain/topic/subtopic-page.model';
+
+class MockWindowRef {
+  nativeWindow = {
+    location: {
+      hostname: 'local'
+    }
+  };
+}
+class MockActiveModal {
+  close(): void {
+    return;
+  }
+
+  dismiss(): void {
+    return;
+  }
+}
+class MockTopicEditorStateService {
+  getTopic() {
+  }
+  getClassroomUrlFragment() {
+    return 'non';
+  }
+  deleteSubtopicPage() {
+  }
+  get onTopicReinitialized(): EventEmitter<void> {
+    let topicReinitializedEventEmitter: EventEmitter<void> = (
+      new EventEmitter());
+    return topicReinitializedEventEmitter;
+  }
+  setSubtopicPage() {
+  }
+}
+
+describe('create new subtopic modal', function() {
+  let component: CreateNewSubtopicModalComponent;
+  let fixture: ComponentFixture<CreateNewSubtopicModalComponent>;
+  let ngbActiveModal: NgbActiveModal;
+  let topicUpdateService: TopicUpdateService;
+  let topicEditorStateService: MockTopicEditorStateService;
+  let subtopicValidationService: SubtopicValidationService;
+  let topicObjectFactory: TopicObjectFactory;
+  let topic = null;
+  let DefaultSubtopicPageSchema = {
+    type: 'html',
+    ui_config: {
+      rows: 100
+    }
+  };
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        CreateNewSubtopicModalComponent
+      ],
+      providers: [
+        {
+          provide: WindowRef,
+          useClass: MockWindowRef
+        },
+        {
+          provide: NgbActiveModal,
+          useClass: MockActiveModal
+        },
+        {
+          provide: TopicEditorStateService,
+          useClass: MockTopicEditorStateService
+        },
+        TopicUpdateService,
+        SubtopicValidationService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateNewSubtopicModalComponent);
+    component = fixture.componentInstance;
+
+    ngbActiveModal = TestBed.inject(NgbActiveModal);
+    TestBed.inject(WindowRef);
+    topicUpdateService = TestBed.inject(TopicUpdateService);
+    topicEditorStateService =
+      (TestBed.inject(TopicEditorStateService) as unknown) as
+      jasmine.SpyObj<MockTopicEditorStateService>;
+    subtopicValidationService = TestBed.inject(SubtopicValidationService);
+    topicObjectFactory = TestBed.inject(TopicObjectFactory);
+
+    topic = topicObjectFactory.createInterstitialTopic();
+    let subtopic1 = Subtopic.createFromTitle(1, 'Subtopic1');
+    topic.getSubtopics = function() {
+      return subtopic1;
+    };
+    topic.getId = function() {
+      return '1';
+    };
+    topic.getNextSubtopicId = function() {
+      return 1;
+    };
+    spyOn(topicEditorStateService, 'getTopic').and.returnValue(topic);
+
+    fixture.detectChanges();
+  });
+
+  it('should assign default values to modal', () => {
```

Comment:
Please follow a 'should do X if Y happens' method to name tests

Reply:
DONE